buildSrc/src/main/java/gradlebuild/JniPlugin.java

```diff
@@ -263,7 +264,13 @@ void createPlatforms(PlatformContainer platformContainer) {
                 // https://github.com/gradle/gradle/blob/36614ee523e5906ddfa1fed9a5dc00a5addac1b0/subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/AbstractGccCompatibleToolChain.java
                 toolChain.target("linux_aarch64");
             });
-            toolChainRegistry.create("clang", Clang.class);
+            toolChainRegistry.create("clang", Clang.class, toolChain -> {
+                // The core Gradle toolchain for Clang only targets x86 and x86_64 out of the box.
+                OperatingSystem os = new DefaultNativePlatform("current").getOperatingSystem();
+                if (os.isMacOsX()) {
+                    toolChain.target("osx_aarch64");
```

Comment:
Does it target `osx_arm64`? I wonder how we determine the default Clang targets. Is that some code in Gradle?

Reply:
No idea, but without this it doesn't work, and with it it does. I guess we can remove the `if` here after https://github.com/gradle/native-platform/pull/266.

Reply:
I think we can remove the `if` when we use a `7.0` Gradle milestone for building native platform itself. We should add `osx_aarch64` as target on non-macOS operating systems.